File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum=nums[0],currSum=0;
        if(nums.size()<2)   return nums[0];
        for(int i=0;i<nums.size();i++){
            currSum+=nums[i];
            maxSum=max(currSum,maxSum);
            if(currSum<0)   currSum=0;
        }
        return maxSum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int currSum=0,maxSum=nums[0];
        if(nums.length<2) return nums[0];
        for(int i:nums){
            currSum+=i;
            maxSum=Math.max(currSum,maxSum);
            if(currSum<0)   currSum=0;
        }
        return maxSum;
    }
}